Frontend/ui-driver/src/Resource/Constants.purs

```diff
@@ -96,3 +98,61 @@ waitTimeConstructor key = case key of
   "Triggered" -> ST.Triggered
   "PostTriggered" -> ST.PostTriggered
   _ -> ST.NoStatus
+
+rideTypeConstructor :: Maybe TripCategory -> ST.TripType
+rideTypeConstructor ( tripCategory) = 
+        case tripCategory of
+            Nothing -> ST.OneWay
+            Just (TripCategory tripCategory') ->
+                case toLower tripCategory'.tag of
+                        "oneway" -> ST.OneWay 
+                        "roundtrip" -> ST.RoundTrip
+                        "rental" -> ST.Rental
+                        "intercity" -> ST.Intercity
+                        "rideshare" -> ST.RideShare
+                        _ -> ST.OneWay
+
+constructLocationInfo :: Maybe Number -> Maybe Number -> Maybe LocationInfo
+constructLocationInfo (latitude) (longitude) = 
+    case latitude,longitude of
+        Just latitude',Just longitude' -> 
+                Just $ LocationInfo {
+                        area :  Just "",
+                        state :  Just "",
+                        country :  Just "",
+                        building :  Just "",
+                        door : Just "",
+                        street :  Just "",
+                        lat : latitude',
+                        city :  Just "",
+                        areaCode :  Just "",
+                        lon : longitude'
+                }
+        _,_ -> Nothing
+
+getLocationInfoFromStopLocation :: StopLocationAddress -> Number -> Number -> LocationInfo
+getLocationInfoFromStopLocation (StopLocationAddress {door, building, street, area, city, state, country, areaCode}) lat lon = 
+     LocationInfo 
+        {
+                area :  area,
+                state :  state,
+                country :  country,
+                building :  building,
+                door : door,
+                street :  street,
+                lat : lat,
+                city :  city,
+                areaCode :  areaCode,
+                lon : lon
+        }
+
+getHomeStageFromString :: String -> ST.HomeScreenStage
+getHomeStageFromString localStage = 
+  case localStage of
+        "HomeScreen" -> ST.HomeScreen
+        "RideRequested" -> ST.RideRequested
+        "RideAccepted" -> ST.RideAccepted
+        "RideStarted" -> ST.RideStarted
+        "RideCompleted" -> ST.RideCompleted
+        "ChatWithCustomer" -> ST.ChatWithCustomer
+        _ -> ST.HomeScreen
```

Comment:
we have multiple screen stages for homeScreen, why rest of them are default to HomeScreen, this can cause unexpected behaviors in future

Reply:
also where are we using it as String? why not directly use the constructors?

Reply:
we are storing the previous homeScreenStage in localstorage and here I was making constructor from string
should i match directly string ?